src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -435,16 +448,20 @@ export class EditDataGridPanel extends GridParentComponent {
 			clearTimeout(this.refreshGridTimeoutHandle);
 
 			this.refreshGridTimeoutHandle = setTimeout(() => {
-
-				if (this.dataSet) {
-					this.placeHolderDataSets[0].dataRows = this.dataSet.dataRows;
-					this.onResize();
-				}
+				try {
+					if (this.dataSet) {
+						this.placeHolderDataSets[0].dataRows = this.dataSet.dataRows;
+						this.onResize();
+					}
 
 
-				if (this.oldDataRows !== this.placeHolderDataSets[0].dataRows) {
-					this.detectChange();
-					this.oldDataRows = this.placeHolderDataSets[0].dataRows;
+					if (this.placeHolderDataSets[0].dataRows && this.oldDataRows !== this.placeHolderDataSets[0].dataRows) {
+						this.detectChange();
+						this.oldDataRows = this.placeHolderDataSets[0].dataRows;
+					}
+				}
+				catch {
+					this.notificationService.error(nls.localize('refreshTableError', 'Too many refreshes, edit session corrupt: Please close and reload table'));
```

Comment:
Do they really have to close and reopen the table? Your description seems to imply that the table will be fine and so shouldn't they just be able to refresh again at that point?

Reply:
The issue is that when you refresh too many times, an error occurs in vscode-jsonrpc module that states an edit session has not been initialized.